### DISCUSSION TITLE
Errors When Connection Settings Are Incorrect

### DISCUSSION THREAD

**Discussion by yuichiromukaiyama** [microsoft/semantic-kernel#5334] [General]
When using the Python version of the semantic kernel, specifying an incorrect environment variable (e.g., a clearly incorrect Azure OpenAI endpoint) does not raise an error. Instead, only an empty string is output.（e.g BasicPlanner）

In production environments, if there is an issue with the connected Azure OpenAI, I would like to output an appropriate error. How do you handle errors in such cases?

**Comment by moonbox3** [comment]
That's a good question, @yuichiromukaiyama. Yes, as you have pointed out, we currently don't have environment variable validation. We do leave it up to the dev/user to make sure their environment variables are well-formed, and only when calling the model would an error occur (most likely a 404). 

What are your thoughts around when an error should be surfaced? The interesting thing is that we wouldn't know an endpoint is invalid until we create the particular client. Only then would we be able to make some "validate" call -- and what would that validate call look like? The model requires the payloads to be formed in a proper way. There doesn't appear to be some "`ping`" or related method to know that the connection can be established properly. At least that I am aware of.
